src/tlo/methods/healthsystem.py

```diff
@@ -1772,6 +1798,52 @@ def call_and_record_never_ran_hsi_event(self, hsi_event, priority=None):
                  priority=priority,
                  )
 
+    def write_to_hsi_no_blank_footprint_log(
+        self,
+        event_details: HSIEventDetails,
+        person_id: int,
+        facility_id: Optional[int],
+        did_run: bool,
+        priority: int,
+    ):
+        """Write the log `HSI_Event` and add to the summary counter only if the appointment footprint is not blank.
+               :param event_details: What event_details is responsible for
+               :param TREATMENT_ID: the treatment ID of the HSI that ran
+               :param Number_By_Appt_Type_Code: dictionary with number of HSIs types
+               :param Person_ID:  A sequence of person ID index values to use as the targets of the HSI events being scheduled
+               :param priority: The priority for the HSI events: 0 (highest), 1 or 2 (lowest). Either a single value for all events or an iterable of per-target values.
+               :param did_run: Boolean asserting if the HSI event did happen
+               :param Facility_Level: The level of the facility where the HSI ran. Can be 0, 1a, 1b, 2, 3, 4, or 5.
+               :param Facility_id: Facility where the HSI ran
+               :param Equipment: The set of equipment that is used in the HSI.
+               """
+        logger.debug(
```

Comment:
why do we need the debug logger?

Reply:
I included it as it was also included in the "write_to_never_ran_hsi_log", but I can remove it